packages/block-library/src/button/index.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * Server-side rendering of the `core/button` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Renders the `core/button` block on the server,
+ *
+ * @since 6.6.0
+ *
+ * @param array    $attributes The block attributes.
+ * @param string   $content    The block content.
+ * @param WP_Block $block      The block object.
+ *
+ * @return string The block content.
+ */
+function render_block_core_button( $attributes, $content ) {
+	$p = new WP_HTML_Tag_Processor( $content );
+
+	/*
+	 * The button block can render an `<a>` or `<button>` and also has a
+	 * `<div>` wrapper. Find the a or button tag.
+	 *
+	 */
+	$tag = null;
+	while ( $p->next_tag() ) {
+		$tag = $p->get_tag();
+		if ( 'A' === $tag || 'BUTTON' === $tag ) {
+			break;
+		}
+	}
+
+	// If this happens, the likelihood is there's no block content.
+	if ( null === $tag ) {
+		return '';
+	}
+
+	// Build a string of the inner text.
+	$text = '';
+	while ( $p->next_token() ) {
+		switch ( $p->get_token_name() ) {
+			case '#text':
+				$text .= $p->get_modifiable_text();
+				break;
+
+			case 'BR':
+				$text .= '';
+				break;
+		}
+	}
+
+	/*
+	 * When there's no text, render nothing for the block.
+	 * It's this way because an anchor or button element without text results
+	 * in poor accessibility.
```

Comment:
does this hold even when there is CSS with the `content` property set? maybe there is additional code involved somewhere else. is removing the content without any indicating the better accessible choice?

Reply:
As mentioned in another comment, it originates from https://github.com/WordPress/gutenberg/issues/17221.

I've updated this comment so that it doesn't mention accessibility, instead I link straight to that issue.